test/end-to-end/company.js

```diff
@@ -1933,4 +1933,294 @@ describe("Company endpoint", () => {
+        });
+
+        afterEach(async () => {
+            await test_agent
+                .delete("/auth/login")
+                .expect(HTTPStatus.OK);
+        });
+
+        afterAll(async () => {
+            await Company.deleteMany({});
+            await Account.deleteMany({});
+        });
+
+        describe("ID Validation", () => {
+            test("Should fail if id is not a valid ObjectID", async () => {
+                await test_agent
+                    .post("/auth/login")
+                    .send(test_user_company)
+                    .expect(HTTPStatus.OK);
+
+                const res = await test_agent
+                    .put("/company/123/edit")
+                    .send()
+                    .expect(HTTPStatus.UNPROCESSABLE_ENTITY);
+
+                expect(res.body.errors[0]).toHaveProperty("msg", ValidationReasons.OBJECT_ID);
+            });
+
+            test("Should fail if id is not a valid company", async () => {
+                const id = "111111111111111111111111";
+                await test_agent
+                    .post("/auth/login")
+                    .send(test_user_company)
+                    .expect(HTTPStatus.OK);
+
+                const res = await test_agent
+                    .put(`/company/${id}/edit`)
+                    .send()
+                    .expect(HTTPStatus.UNPROCESSABLE_ENTITY);
+
+                expect(res.body.errors[0]).toHaveProperty("msg", ValidationReasons.COMPANY_NOT_FOUND(id));
+            });
+        });
+
+        describe("Should fail if bad user", () => {
+            test("Should fail if different user", async () => {
+                await test_agent
+                    .post("/auth/login")
+                    .send(test_random_user)
+                    .expect(HTTPStatus.OK);
+
+                const res = await test_agent
+                    .put(`/company/${test_company._id}/edit`)
+                    .send({
+                        name: "Changing Company",
+                        bio: "Without permission",
+                        contacts: ["1"],
+                        logo: "http://awebsite.com/otherlogo.jpg",
+                    })
+                    .expect(HTTPStatus.FORBIDDEN);
+
+                expect(res.body.errors[0]).toHaveProperty("msg", ValidationReasons.INSUFFICIENT_PERMISSIONS_COMPANY_SETTINGS);
+            });
+
+            test("Should fail if not logged in", async () => {
+                const res = await test_agent
+                    .put(`/company/${test_company._id}/edit`)
+                    .send({
+                        name: "Changing Company",
+                        bio: "Without login",
+                        contacts: ["1"],
+                        logo: "http://awebsite.com/otherlogo.jpg",
+                    })
+                    .expect(HTTPStatus.UNAUTHORIZED);
+
+                expect(res.body.errors[0]).toHaveProperty("msg", ValidationReasons.INSUFFICIENT_PERMISSIONS);
```

Comment:
We normally use a `toContainEqual`: just a suggestion.

Reply:
Fixed this!